commit b7a080bad81acb88e28b393a466fdadbbb406d4f (refs/changes/77/2677/4)
Author: Pedro Palazon Candel <pedro@joyent.com>
Date:   2017-10-04T14:41:19+00:00 (2 years ago)
    
    TOOLS-1870 sdcadm update <agent> timeout needs to be bigger than 5 min
    Reviewed by: Julien Gilli <julien.gilli@joyent.com>
    Approved by: Julien Gilli <julien.gilli@joyent.com>

diff --git a/lib/procedures/update-agent-v1.js b/lib/procedures/update-agent-v1.js
index 587387c..385af80 100644
--- a/lib/procedures/update-agent-v1.js
+++ b/lib/procedures/update-agent-v1.js
@@ -259,12 +259,13 @@ UpdateAgentV1.prototype.execute = function uaExecute(opts, callback) {
                 // anything but cn-agent:
                 function waitUntilTaskCompletes(taskid, _cb) {
                     var counter = 0;
-                    var limit = 60;
+                    var limit = 360;
                     function _waitTask() {
                         counter += 1;
                         sdcadm.cnapi.getTask(taskid, function (err, task) {
                             if (err) {
-                                return _cb(new SDCClientError(err, 'cnapi'));
+                                _cb(new SDCClientError(err, 'cnapi'));
+                                return;
                             }
 
                             if (task.status === 'failure') {
@@ -273,16 +274,16 @@ UpdateAgentV1.prototype.execute = function uaExecute(opts, callback) {
                                     msg += ' with error: ' +
                                         task.history[0].event.error.message;
                                 }
-                                return _cb(new UpdateError(msg));
+                                _cb(new UpdateError(msg));
                             } else if (task.status === 'complete') {
-                                return _cb();
+                                _cb();
                             } else if (counter < limit) {
-                                return setTimeout(_waitTask, 5000);
+                                setTimeout(_waitTask, 5000);
                             } else {
                                 var message = format(
-                                    'Timeout(5m) waiting for task %s', taskid);
+                                    'Timeout(30m) waiting for task %s', taskid);
                                 progress(message);
-                                return _cb(new UpdateError(message));
+                                _cb(new UpdateError(message));
                             }
                         });
                     }
@@ -293,13 +294,14 @@ UpdateAgentV1.prototype.execute = function uaExecute(opts, callback) {
                 // when we are updating cn-agent
                 function waitUntilAgentsChange(server_uuid, _cb) {
                     var counter = 0;
-                    var limit = 60;
+                    var limit = 360;
                     function _waitServer() {
                         counter += 1;
                         sdcadm.cnapi.getServer(server_uuid,
                                 function (err, server) {
                             if (err) {
-                                return _cb(new SDCClientError(err, 'cnapi'));
+                                _cb(new SDCClientError(err, 'cnapi'));
+                                return;
                             }
                             var theAgent = server.agents.filter(
                                     function (a) {
@@ -307,15 +309,15 @@ UpdateAgentV1.prototype.execute = function uaExecute(opts, callback) {
                             })[0];
 
                             if (theAgent.image_uuid === change.image.uuid) {
-                                return _cb();
+                                _cb();
                             } else if (counter < limit) {
-                                return setTimeout(_waitServer, 5000);
+                                setTimeout(_waitServer, 5000);
                             } else {
-                                var msg = format('Timeout(5m) waiting for ' +
+                                var msg = format('Timeout(30m) waiting for ' +
                                         'cn-agent update on server %s',
                                         server_uuid);
                                 progress(msg);
-                                return _cb(new UpdateError(msg));
+                                _cb(new UpdateError(msg));
                             }
                         });
                     }
@@ -432,7 +434,7 @@ UpdateAgentV1.prototype.execute = function uaExecute(opts, callback) {
 
 };
 
-//---- exports
+// --- exports
 
 module.exports = {
     UpdateAgentV1: UpdateAgentV1
